cmd/krew/cmd/util.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"sigs.k8s.io/krew/pkg/constants"
+	"sigs.k8s.io/krew/pkg/index"
+)
+
+// indexOf returns the index name of a receipt.
+func indexOf(r index.Receipt) string {
+	if r.Status.Source.Name == "" {
+		return constants.DefaultIndexName
+	}
+	return r.Status.Source.Name
+}
+
+// displayName returns the display name of a Plugin.
+// The index name is omitted if it is the default index.
+func displayName(p index.Plugin, indexName string) string {
```

Comment:
Should this just have `index.Receipt` as its parameter? I'm guessing you added it this way for this to be more generic/to be able to get the display name when a receipt doesn't exist.

Reply:
Yeah, I added this so I can make it work w/ Plugin (in "krew search" patch I have). It's easily working w/ Receipts right now; but once we accept Receipt, we can't make it work w/ Plugin. 

Also conceptually this is a "Plugin" concern. Just because receipt has a source index name, don't feel obliged to pass it around everywhere. :)